src/app/app.component.ts

```diff
@@ -1,9 +1,68 @@
 import { Component } from '@angular/core';
+import { angularMaterialRenderers } from '@jsonforms/angular-material';
+import { and, isControl, optionIs, rankWith, schemaTypeIs, scopeEndsWith, Tester } from '@jsonforms/core';
+import { CustomAutocompleteControlRenderer } from './custom.autocomplete';
+import { DataDisplayComponent } from './data.control';
+import { LangComponent } from './lang.control';
+import uischemaAsset from '../assets/uischema.json';
+import schemaAsset from '../assets/schema.json';
+import dataAsset from './data';
+import AJV from 'ajv';
+import { parsePhoneNumber } from 'libphonenumber-js';
+
+const departmentTester: Tester = and(
+  schemaTypeIs('string'),
+  scopeEndsWith('department')
+);
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
 export class AppComponent {
+  renderers = [
+    ...angularMaterialRenderers,
+    { tester: rankWith(5, departmentTester), renderer: CustomAutocompleteControlRenderer },
+    {
+      renderer: DataDisplayComponent,
+      tester: rankWith(
+        6,
+        and(
+          isControl,
+          scopeEndsWith('___data')
+        )
+      )
+    },
+    {
+      renderer: LangComponent,
+      tester: rankWith(
+        6,
+        and(
+          isControl,
+          optionIs('lang', true)
+        )
+      )
+    },
+  ];
+  uischema = uischemaAsset;
+  schema = schemaAsset;
+  data = dataAsset;
+  ajv = new AJV({
+    schemaId: 'auto',
+    allErrors: true,
+    jsonPointers: true,
+    errorDataPath: 'property'
+  });
```

Comment:
I think we should use `createAjv` which is exported from JSON Forms as it automatically sets all options we need.

Reply:
good catch, I updated